compose-guard/gradle-plugin/src/main/kotlin/com/joetr/compose/guard/ComposeChecks.kt

```diff
@@ -139,4 +141,51 @@ public object ComposeChecks {
             )
         }
     }
+
+    private fun checkUnstableParams(
+        checkedMetrics: ComposeCompilerMetricsProvider,
+        goldenMetrics: ComposeCompilerMetricsProvider,
+    ) {
+        val checkedUnstableParamsMap = mutableMapOf<ComposableDetail.FunctionAndParameter, String>()
+        checkedMetrics.getComposablesReport().composables.forEach { composable ->
+            composable.params.filter {
+                it.condition == UNSTABLE
+            }.forEach { param ->
+                checkedUnstableParamsMap[
+                    ComposableDetail.FunctionAndParameter(
+                        functionName = composable.functionName,
+                        parameterName = param.name,
+                        parameterType = param.type,
+                    ),
+                ] = param.stabilityStatus.toString()
+            }
+        }
+
+        val goldenUnstableParamsMap = mutableMapOf<ComposableDetail.FunctionAndParameter, String>()
+        goldenMetrics.getComposablesReport().composables.forEach { composable ->
+            composable.params.filter {
+                it.condition == UNSTABLE
+            }.forEach { param ->
+                goldenUnstableParamsMap[
+                    ComposableDetail.FunctionAndParameter(
+                        functionName = composable.functionName,
+                        parameterName = param.name,
+                        parameterType = param.type,
+                    ),
+                ] = param.stabilityStatus.toString()
+            }
+        }
+
+        // find new unstable parameters by removing all golden parameters from checked parameters
+        checkedUnstableParamsMap.keys.removeAll(goldenUnstableParamsMap.keys)
+
+        if (checkedUnstableParamsMap.isNotEmpty()) {
+            throw GradleException(
+                "New unstable parameters were added! \n" +
+                    checkedUnstableParamsMap.keys.joinToString(separator = ",") + "\n" +
+                    "More info: https://github.com/androidx/androidx/blob/androidx-main/compose/compiler/design/compiler-metrics.md" +
+                    "#parameters-that-are-unstable",
```

Comment:
this url doesn't not appear to be valid :) all of the other ones are actual anchors on that page